contrib/generic_proxy/filters/network/source/interface/codec.h

```diff
@@ -178,14 +112,50 @@ class CodecFactory {
 
 using CodecFactoryPtr = std::unique_ptr<CodecFactory>;
 
+class FilterConfig;
+using FilterConfigSharedPtr = std::shared_ptr<FilterConfig>;
+
+/**
+ * Custom read filter factory for generic proxy.
+ */
+class FilterFactory {
+public:
+  virtual ~FilterFactory() = default;
+
+  /**
+   * Create a custom read filter instance.
+   * @param filter_manager the filter manager of the network filter chain.
+   * @param filter_config supplies the filter config.
+   */
+  virtual void createFilter(Network::FilterManager& filter_manager,
+                            const FilterConfigSharedPtr& filter_config) const PURE;
+};
+using FilterFactoryPtr = std::unique_ptr<FilterFactory>;
+
 /**
  * Factory config for codec factory. This class is used to register and create codec factories.
  */
 class CodecFactoryConfig : public Envoy::Config::TypedFactory {
 public:
+  /**
+   * Create a codec factory. This should never return nullptr.
+   * @param config supplies the config.
+   * @param context supplies the server context.
+   * @return CodecFactoryPtr the codec factory.
+   */
   virtual CodecFactoryPtr createFactory(const Protobuf::Message& config,
                                         Envoy::Server::Configuration::FactoryContext& context) PURE;
 
+  /**
+   * Create a optional custom filter factory.
+   * @param config supplies the config.
+   * @param context supplies the server context.
+   * @return FilterFactoryPtr the filter factory.
+   */
+  virtual FilterFactoryPtr
+  filterFactory(const Protobuf::Message& config,
```

Comment:
There is a little confuse on the name of this two factory method: 'createFactory' and 'filterFactory'. Maybe one is createCodecFactory, another one is createCustomFilterFactory?

Reply:
Or createCustomGenericProxyFilterFactory, the `filter` is really confused since we have too many filters.

Reply:
used `createCodecFactory` and `createCustomGenericProxyFilterFactory`.